nautobot/extras/models/jobs.py

```diff
@@ -521,27 +522,57 @@ class JobResult(BaseModel, CustomFieldModel):
     )
 
     name = models.CharField(max_length=255, db_index=True)
+    task_name = models.CharField(max_length=255, null=True, db_index=True)
+
     obj_type = models.ForeignKey(
         to=ContentType,
         related_name="job_results",
         verbose_name="Object types",
         limit_choices_to=FeatureQuery("job_results"),
         help_text="The object type to which this job result applies",
         on_delete=models.CASCADE,
+        null=True,
+        blank=True,
     )
-    created = models.DateTimeField(auto_now_add=True)
-    completed = models.DateTimeField(null=True, blank=True)
+
+    date_created = models.DateTimeField(auto_now_add=True)
+    date_done = models.DateTimeField(null=True, blank=True)
     user = models.ForeignKey(
         to=settings.AUTH_USER_MODEL, on_delete=models.SET_NULL, related_name="+", blank=True, null=True
     )
-    # todoindex:
     status = models.CharField(
         max_length=30,
         choices=JobResultStatusChoices,
         default=JobResultStatusChoices.STATUS_PENDING,
+        help_text="Current state of the Job being run",
+        db_index=True,
     )
     data = models.JSONField(encoder=DjangoJSONEncoder, null=True, blank=True)
-    job_kwargs = models.JSONField(blank=True, null=True, encoder=NautobotKombuJSONEncoder)
+    periodic_task_name = models.CharField(null=True, max_length=255)
+    worker = models.CharField(max_length=100, default=None, null=True)
+    task_args = models.JSONField(blank=True, null=True, encoder=NautobotKombuJSONEncoder)
+    task_kwargs = models.JSONField(blank=True, null=True, encoder=NautobotKombuJSONEncoder)
+    content_type = models.CharField(
+        default="application/x-nautobot-json",
+        max_length=128,
+        verbose_name=("Result Content Type"),
+        help_text="Content type of the result data",
+    )
+    content_encoding = models.CharField(
+        default="utf-8",
+        max_length=64,
+        verbose_name=("Result Encoding"),
+        help_text="The encoding used to save the task result data",
+    )
+    result = models.TextField(
+        null=True,
+        default=None,
+        editable=False,
+        verbose_name="Result Data",
+        help_text="The data returned by the task. Use content_encoding and content_type fields to read.",
+    )
+    traceback = models.TextField(blank=True, null=True)
+    meta = models.JSONField(null=True, default=None, editable=False)
```

Comment:
Surprising to me that there isn't an abstract base class we can inherit from for this set of fields.

Reply:
That would be lovely. Alas: https://github.com/celery/django-celery-results/blob/47402b8120b7b84e741b1a38d67caf05146c62a9/django_celery_results/models.py#L18